iceoryx_dust/container/include/iox/detail/fixed_position_container_helper.hpp

```diff
@@ -17,81 +17,146 @@
 #ifndef IOX_DUST_CONTAINER_DETAIL_FIXED_POSITION_CONTAINER_HELPER_HPP
 #define IOX_DUST_CONTAINER_DETAIL_FIXED_POSITION_CONTAINER_HELPER_HPP
 
+#include <functional>
 #include <utility>
 
 namespace iox
 {
 namespace detail
 {
-template <bool IsMove>
-struct AssignmentHelper;
 
-template <>
-struct AssignmentHelper<true>
+enum class SpecialCreationOperations
 {
-    template <typename T>
-    static void assign(T& dest, T&& src)
-    {
-        dest = std::forward<T>(src);
-    }
+    CopyConstructor,
+    CopyAssignment,
+    MoveConstructor,
+    MoveAssignment,
 };
 
-template <>
-struct AssignmentHelper<false>
+/// @brief SpecialCreationHelper is a template structure used to create or assign objects based on the provided
+/// operation type (Opt).
+/// @tparam Opt The operation type that determines how objects are created or assigned.
+template <SpecialCreationOperations Opt>
+struct SpecialCreationHelper
 {
-    template <typename T>
-    static void assign(T& dest, const T& src)
+    /// @brief Creates or assigns an object to 'dest' based on the specail operation type.
+    /// @tparam T The type of the object to be created or assigned.
+    /// @tparam V The type of the source object, kept as a universal reference to preserve its lvalue or rvalue nature.
+    /// @param[out] dest The destination object where the new object is created or to which the source object is
+    /// assigned.
+    /// @param[in] src The source object, either for copy or move operations.
+    template <typename T, typename V>
+    static inline void create(T& dest, V&& src) noexcept
```

Comment:
This name is also a bit misleading since it may also assign. Maybe we find a more fitting name.

Reply:
Indeed, finding a single term that encapsulates both construction and assignment is quite challenging :(

While `construct` seems like a suitable name, it could potentially mislead users into thinking the function only deals with object construction. I initially chose `create` precisely because it broadly suggests the creation of a new instance, encompassing both construction and assignment functionalities.

Do you have any idea?

Reply:
Not really. The only thing I can think of is `copy_or_move`. This shouldn't block the PR though. We can take care of the name when the helper struct is moved to `iceoryx_hoofs/design`

Reply:
I'll use `transfer` as a temp name.